sign/rpmgensig.c

```diff
@@ -276,7 +282,26 @@ static int runGPG(sigTarget sigt, const char *sigfile)
     rpmPopMacro(NULL, "__plaintext_filename");
     rpmPopMacro(NULL, "__signature_filename");
 
+    /* The child GPG process may terminate without ever opening the pipe (such
+     * as when the key is expired), causing our parent process to get forever
+     * stuck on the Fopen() call below (see fifo(7) for details).  To fix that,
+     * we need to let the underlying open(2) unblock when SIGCHLD is received,
+     * by registering a no-op handler, so here we go: */
+    struct sigaction act;
+    memset(&act, '\0', sizeof(act));
```

Comment:
memset() second argument is an int, not a character. Which seems kinda bizarre if you start thinking about bytes and all, but that's the way it is, and so the argument should be just plain zero, not the nul character. Not that it makes any difference functionality-wise.

Reply:
Oh OK, thanks for pointing this out, I'll fix this.